crates/l3/common/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+
+type RpcAccount<'a> = SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>;
+pub type TransactionExecution<'a> = Execution<'a, RpcAccount<'a>>;
+
+pub const NO_CONSTRUCTOR_ARG: Vec<FieldElement> = Vec::new();
+
+pub trait StarknetContractClient {
+    fn address(&self) -> FieldElement;
+    fn client(&self) -> LocalWalletSignerMiddleware;
+}
+
+
+
+
+pub async fn invoke_contract<'a>(
+    client: &'a Arc<LocalWalletSignerMiddleware>,
+    address: FieldElement,
+    method: &str,
+    calldata: Vec<FieldElement>,
+) -> Execution<'a, LocalWalletSignerMiddleware> {
+    let call = Call {
+        to: address,
+        selector: get_selector_from_name(method.into()).unwrap(),
+        calldata: calldata,
+    };
+    client.as_ref().execute(vec![call])
+}
+
+// pub async fn call_contract(
+//     client: Arc<LocalWalletSignerMiddleware>,
+//     address: FieldElement,
+//     method: &str,
+// ) -> Option<Vec<FieldElement>> {
+//     let function_call = FunctionCall {
+//         contract_address: address,
+//         entry_point_selector: get_selector_from_name(method.into()).unwrap(),
+//         calldata: vec![],
+//     };
+//     let provider = client.provider();
+//     match provider
+//         .call(function_call, BlockId::Tag(BlockTag::Latest))
+//         .await
+//     {
+//         Ok(result) => Some(result),
+//         Err(_) => None,
+//     }
+// }
+
+pub async fn call_contract(
+    client: &LocalWalletSignerMiddleware,
+    address: FieldElement,
+    method: &str,
+) -> Result<Vec<FieldElement>, Error> {
+    let function_call = FunctionCall {
+        contract_address: address,
+        entry_point_selector: get_selector_from_name(method.into()).unwrap(),
+        calldata: vec![],
+    };
+    let provider = client.provider();
+    provider
+        .call(function_call, BlockId::Tag(BlockTag::Latest))
+        .await
+        .map_err(|e| Error::CustomError(format!("Failed to call contract: {}", e)))
+        .map(|result| result)
+}
+
+pub async fn deploy_contract(
+    client: Arc<LocalWalletSignerMiddleware>,
+    path_to_sierra: &str,
+    path_to_casm: &str,
+    constructor_args: Vec<FieldElement>,
+) -> Result<FieldElement, Error> {
+    let sierra_file = std::fs::File::open(path_to_sierra)
+        .map_err(|e| Error::CustomError(format!("Failed to open Sierra file: {}", e)))?;
+    let sierra: SierraClass = serde_json::from_reader(sierra_file)?;
+    let casm_file = std::fs::File::open(path_to_casm)
+        .map_err(|e| Error::CustomError(format!("Failed to open CASM file: {}", e)))?;
+    let casm: CompiledClass = serde_json::from_reader(casm_file)?;
+    let compiled_class_hash = casm.class_hash()
+        .map_err(|e| Error::CustomError(format!("Failed to get class hash from CASM: {}", e)))?;
+    let declare_tx = client
+        .declare(
+            sierra.clone().flatten()
+                .map_err(|e| Error::CustomError(format!("Failed to flatten Sierra class: {}", e)))?.into(),
```

Comment:
instead of using Custom Errors everywhere, you should check the color_eyre package, it should be the correct use case over here

Reply:
@apoorvsadana 
How should this be implemented, since when these errors then will be required to propagated to change across all the methods where it is called and will be called from other repo as well. 
Or should we create a enum Error with all specific errors and use that?

Reply:
So I was thinking of using eyre errors in those places where we were using CustomErrors, I guess we replaced all of them? Anyways, I think this works as well, later when we need to handle any error specifically, we can see what to do.

Reply:
done